lib/core/src/Cardano/Wallet/Api/Helpers.hs

```diff
@@ -0,0 +1,80 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveFunctor #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE StandaloneDeriving #-}
+
+module Cardano.Wallet.Api.Helpers
+    ( ApiT (..)
+    , eitherToParser
+    , hexText
+    , fromHexText
+    , toTextJSON
+    , fromTextJSON
+    )
+    where
+
+import Prelude
+
+import Cardano.Wallet.Primitive.AddressDerivation
+    ( fromHex, hex )
+import Cardano.Wallet.Util
+    ( ShowFmt (..) )
+import Control.DeepSeq
+    ( NFData )
+import Data.Aeson
+    ( ToJSON (..), Value, withText )
+import Data.Bifunctor
+    ( Bifunctor (..) )
+import Data.ByteString
+    ( ByteString )
+import Data.Hashable
+    ( Hashable )
+import Data.Text
+    ( Text )
+import Data.Text.Class
+    ( FromText (..), ToText (toText) )
+import GHC.Generics
+    ( Generic )
+import Quiet
+    ( Quiet (..) )
+
+import qualified Data.Aeson.Types as Aeson
+import qualified Data.Text.Encoding as T
+
+-- | Polymorphic wrapper type to put around primitive types and, 3rd party lib
+-- types to avoid defining orphan instances and/or, undesirable instances on
+-- primitive types. It helps to keep a nice separation of concerns between the
+-- API layer and other modules.
+newtype ApiT a =
+    ApiT
+    { getApiT :: a
+    }
+    deriving ( Generic, Eq, Functor )
+    deriving newtype ( Semigroup, Monoid, Hashable )
+    deriving anyclass NFData
+    deriving Show via (Quiet (ApiT a))
+
+deriving instance Ord a
+    => Ord (ApiT a)
```

Comment:
I suggest to extract the `Cardano.Wallet.Api.Trans` module and place it there.

Reply:
factoring it out is the only way we can split the monster

Reply:
So you're trying to solve the problem of a "monster"?

In which case lets solve it in a more systematic way:
1. Define what the problem exactly is. 
2. Outline possible solutions.
3. Pick the best solution.

Personally I struggle to define the problem here.  What exactly concerns you? 
- the fact that the file has many lines? Why? We have many huge files and it didn't seem to bother developers there;
- the fact that file module file has many instances? Why is it a problem again?

Reply:
Many lines implies long compilation times, I think this was clear that we do not want to make  them bigger at least

Reply:
I moved back the `ApiT` type.